Feature/59 add dependency container

HijoDeHonor opened the pull request:
Added a dependency container ussing awilix to manage and inject dependencies throughout the application for improved modularity and maintainability.

HijoDeHonor committed:
feat: add dependency container

Added a dependency container to manage and inject dependencies throughout the application for improved modularity and maintainability.

HijoDeHonor committed:
fix: adjust tests for dependency injection

Corrected tests to properly utilize dependency injection, ensuring accurate test behavior and reliability.

PiewDev commented on dependencyInjection/di-settings.js:
```diff
@@ -0,0 +1,20 @@
+});
+
+export function settings () {
+  container.register({
```
don't use singleton, use scoped

PiewDev commented on errors/errorHandler/errorhandler.js:
```diff
@@ -4,5 +4,6 @@ import { errors } from './errorDictionary.js';
 export const errorHandler = (error, req, res, next) => {
   const statusCode = errors[error.name] || 500;
   const message = statusCode >= 500 ? INTERNAL_SERVER_ERROR : error.message;
+  console.log(error);
```
remove the log

PiewDev commented on index.js:
```diff
@@ -4,33 +4,27 @@ import cookieParser from 'cookie-parser';
 app.use(express.json());
 app.use(cookieParser());
-//app.use(corsMiddelware())
+// app.use(corsMiddelware())
```
delete this comment

PiewDev reviewed (changes requested):
There are some commented changes to make, to that we are going to add separating the tests by functionality: 

-test
--user
---login
----loginUserController.test.js
----loginUserRepository.test.js
----...
---Register
----...

HijoDeHonor committed:
refactor: reorganize tests, remove comments, update Awilix container

- Separated tests by use case for better organization
- Removed unnecessary comments for cleaner code
- Changed methods from singleton to scoped in the Awilix container for improved dependency management

PiewDev reviewed (approved):
Good Job!
